### ISSUE TITLE
[Idea] Splitting the bottom pane

### ISSUE THREAD

**Issue by Mrnofish** [xavery/transgui#27]
The bottom pane currently holds not that much information and fails to meaningfully fill up even a regular FHD monitor, there's literally tons of unused real estate.

Instead of forcing the user to hop from tab to tab, the bottom pane could be split vertically in two (at least when the application is maximized), and allow showing two tabs at the same time.

Here's a rough mock up that illustrates the basic idea.

![transgui_doublepane_mockup](https://github.com/xavery/transgui/assets/28036193/fd102301-5c29-479f-89a5-06b320424190)

Instead of having redundant tab controls, as there would be no point in having the same tab showing twice, it could support splitting / unsplitting the bottom pane with a mechanism similar to the Other View in Notepad++, so that the user can choose whether to display two tabs or just one as it is now.

Right clicking on a Tab's name would allow sending tabs from one view to the other, and let the user customize which two tabs to display.

I'm not sure how much work this would entail, or whether it falls in the scope of the project, so just a suggestion to perhaps get the ball rolling.

**Comment by xavery** [comment]
Thanks for the report. I can see the reasoning though I personally don't experience it as my desktop's DPI is 112 so the layout is quite different and it just doesn't look so empty.

I guess it would be the easiest to just start out with a vertical splitter and two identical UI components on both sides. That would have the functionality you're describing (although the duplication would be kind of confusing) but messing around with tabs, let alone docking/undocking, is probably far beyond what I'm capable of here.

**Comment by Mrnofish** [comment]
Yeah the duplication would be not look pretty, the most basic way to go about that would be hard-coding the tab setup, distributing them between the two panes in a fixed fashion.

For instance, have only General and Trackers on the Left, and Peers, Files and Statistics on the Right only.

The ones that are used the most should be Files, Trackers and Peers, more or less in the order, so that arrangement should at least allievate some tab hopping.

FWIW Statistics is mostly useless and should have never been placed in the bottom pane at all since it only prints data for the entire Transmission set up and not the individual torrent.

>  I can see the reasoning though I personally don't experience it as my desktop's DPI is 112 so the layout is quite different and it just doesn't look so empty.

On a FHD laptop screen you may not run into that issue, however for desktop computers, the standard monitor size has been 24 in for a long time now, and many, if not most, can live with FHD at that size without having to increase the UI scale.

A few years ago it was also not infrequent for folks with less than perfect eyesight to choose FHD 25 or 27 inchers. But 1440p 27 and 32 inchers can also be used without scaling.

UHD is a different kettle of fish and pretty much requires scaling until you hit about 40 - 43in, at which point you have about the same DPI as 27" 1440p screen. That's typically a TV screen used as a monitor, although there are computer screens that size, too.

**Comment by lighterowl** [comment]
Although it may not seem like it, stuff is happening. This is obviously very, very far from being called finished but getting even such a small part of this working was encouraging as it shows that this can be done.

https://github.com/xavery/transgui/assets/127875/77152196-34bc-4ac2-b589-5027847cd380
